pkg/injector/pod_patch.go

```diff
@@ -318,6 +319,29 @@ func getMaxRequestBodySize(annotations map[string]string) (int32, error) {
 	return getInt32Annotation(annotations, daprMaxRequestBodySize)
 }
 
+// add env-vars from annotations.
+// see https://github.com/dapr/dapr/issues/2508.
+func getEnvVarsAnnotation(annotations map[string]string) []corev1.EnvVar {
+	envVars := make([]corev1.EnvVar, 0)
+	envStr := annotations[daprEnvKey]
+	envPairs := strings.Split(envStr, ",")
+
+	for _, value := range envPairs {
+		pair := strings.Split(strings.Trim(value, " "), "=")
+
+		if len(pair) != 2 {
+			continue
```

Comment:
@yaron2 in this scenario, should it be logged as warn?

Reply:
It is OK not to log anything here IMO. I actually think this code should be in an utility package.

Reply:
@artursouza  I have moved the `getEnvVarsAnnotation` function to util as named `parseEnvString`.